core/src/main/java/org/apache/calcite/rel/core/Window.java

```diff
@@ -111,6 +111,14 @@ public Window(RelOptCluster cluster, RelTraitSet traitSet, RelNode input,
     this(cluster, traitSet, Collections.emptyList(), input, constants, rowType, groups);
   }
 
+  /**
+   * Creates a copy of this {@code Window}.
+   *
+   * @param constants List of constants that are additional inputs
+   * @return New {@code Window}
+   */
+  public abstract Window copy(List<RexLiteral> constants);
```

Comment:
I would actually call this method withConstants, in line with other naming conventions in Calcite.
I find the JavaDoc confusing, because these are not *additional* inputs, these replace the existing constants of the window.

Reply:
I agree with the confusing javadoc.

I have no strong opinion on the method name, but I have the impression that, in terms of RelNode "operators", in other similar cases we have named them `copy` .

Independently of the name, we could consider this a "breaking change" (any downstream project extending Window will be "broken" on the next Calcite upgrade, and will require a minor adjustment to provide the implementation of this new abstract method). So I'd propose to add the corresponding comment in `history.md` (in the commented out "breaking changes" section for next release).

Reply:
I can rename this, but `copy` is also used like this in other classes.

https://github.com/apache/calcite/blob/main/core/src/main/java/org/apache/calcite/rel/core/Filter.java#L125
https://github.com/apache/calcite/blob/main/core/src/main/java/org/apache/calcite/rel/logical/LogicalFilter.java#L148

Should I still rename this method?

Reply:
if that's the convention you can keep it this way.

Reply:
I've updated the Javadoc comment to better describe how `constants` is used.

Reply:
I've added a comment to the API changes for 1.37.0 in the `history.md` file.